folium/features.py

```diff
@@ -767,12 +795,20 @@ class DivIcon(MacroElement):
     def __init__(self, html=None, icon_size=None, icon_anchor=None,
                  popup_anchor=None, class_name='empty'):
         super(DivIcon, self).__init__()
-        self._name = 'DivIcon'
+        self._name = 'divIcon'
         self.icon_size = icon_size
         self.icon_anchor = icon_anchor
         self.popup_anchor = popup_anchor
         self.html = html
         self.className = class_name
+        self.options = {
```

Comment:
I like that you're parsing these options in Python instead of the template, but the template edit is missing. Also, this should be a separate PR.

Reply:
Fair enough. My goal is to have an options dict that gets `json.dump`ed for each class eventually to easily drop into the template callback, in lieu of hard coding each value. This will play nicer with **kwargs too. In a few of these I went ahead and created the dict, but didn't updated the template logic yet.